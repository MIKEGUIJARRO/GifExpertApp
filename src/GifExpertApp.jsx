import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) {
      return;
    }

    setCategories((prevValues) => {
      return [newCategory, ...prevValues];
    });
  };

  return (
    <div className="p-16 space-y-4">
      <h1 className="text-5xl font-bold">Gif Export App</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </div>
  );
};
