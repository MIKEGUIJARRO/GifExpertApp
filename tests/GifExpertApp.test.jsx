import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Test in <GifExpertApp />', () => {

    test('should match the snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
        //screen.debug();
    });

    test('should show the h1 title', () => {
        render(<GifExpertApp />);
        expect(screen.getByText('Gif Export App'));
        //screen.debug();
    });

    test('should add a category title when form is filled and sent and in loading state', () => {
        const inputValue = 'One Punch Man';
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue } });
        fireEvent.submit(form);
        
        expect(screen.getByText(inputValue));
        expect(screen.getByText('Is loading...'));
    });
});