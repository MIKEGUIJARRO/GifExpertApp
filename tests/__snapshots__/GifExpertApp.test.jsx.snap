// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp /> should 1`] = `
<div>
  <div
    class="p-16 space-y-4"
  >
    <h1
      class="text-5xl font-bold"
    >
      Gif Export App
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="bg-gray-200 px-2 py-1 rounded-md"
        placeholder="Buscar Gifs"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;

exports[`Test in <GifExpertApp /> should match the snapshot 1`] = `
<div>
  <div
    class="p-16 space-y-4"
  >
    <h1
      class="text-5xl font-bold"
    >
      Gif Export App
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="bg-gray-200 px-2 py-1 rounded-md"
        placeholder="Buscar Gifs"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
